hypothesis-python/tests/cover/test_reflection.py

```diff
@@ -346,6 +350,13 @@ def test_uses_varargs():
     assert f(1, 2) == ((1, 2), {})
 
 
+def test_copying_preserves_coroutines():
+    f = define_function_signature("foo", "A docstring for foo", signature(has_varargs))(
+        async_acceptor
+    )
```

Comment:
let's do 
```python
@given(st.none())
async def f(x): ...
```

to make it clear that this is reachable via public interfaces only

Reply:
`@given` on an async function doesn't actually encounter `define_function_signature`, fwiw. (that would be pretty concerning for our support for pytest-asyncio and friends!)